torch/csrc/api/include/torch/data/htk/htk_parser.hpp

```diff
@@ -0,0 +1,175 @@
+#pragma once
+#include <vector>
+#include <iostream>
+#include <fstream>
+#include <cassert>
+#include <optional>
+
+#include <cstddef>
+#include <typeinfo>
+
+using namespace std;
+
+#define _DEBUG 1
+
+struct Utterance
+{
+  std::vector<float> feature;
+  std::vector<int32_t> label;
+  std::vector<uint8_t> lattice;
+};
+
+template <typename T>
+T swap_endian(T u)
+{
+  union {
+    T u;
+    unsigned char u8[sizeof(T)];
+  } source, dest;
+
+  source.u = u;
+
+  for (size_t k = 0; k < sizeof(T); k++)
+    dest.u8[k] = source.u8[sizeof(T) - k - 1];
+
+  return dest.u;
+}
+
+class HTKParser
+{
+public:
+  using FeatureType = float;
```

Comment:
Can this be fixed like this? or do we need them to be flexible?

Reply:
I don't see any place in HTK data file that indicates data types. These types I got it from Mudit's script.